python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -1334,10 +1336,20 @@ def mult(df):
         1  2     1
         2  3     1
         """
+
+        if self.obj.empty:
+            return self.obj
         if not callable(function):
             raise TypeError(f"type {type(function)} is not callable")
         group_names, offsets, group_keys, grouped_values = self._grouped()
 
+        if engine == "auto":
+            if (not grouped_values._has_nulls) and _can_be_jitted(
```

Comment:
Should this null check be part of the definition of `_can_be_jitted`?

Reply:
I kept it separate here because it's only the groupby pipeline that doesn't have null support. If we wanted to use this function for basic `apply` we'd want it to pass through frames with nulls.